x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/tool/BaseRunAsSuperuserCommand.java

```diff
@@ -0,0 +1,281 @@
+
+            attributesChecker.check(terminal);
+            checkClusterHealthWithRetries(newEnv, terminal, 5);
+            executeCommand(terminal, options, newEnv);
+        } catch (Exception e) {
+            int exitCode;
+            if (e instanceof UserException) {
+                exitCode = ((UserException) e).exitCode;
+            } else {
+                exitCode = ExitCodes.DATA_ERROR;
+            }
+            throw new UserException(exitCode, e.getMessage());
+        } finally {
+            cleanup(terminal, newEnv);
+        }
+    }
+
+    /**
+     * Removes temporary file realm user from users and roles file
+     */
+    protected void cleanup(Terminal terminal, Environment env) throws Exception {
+        final Path passwordFile = FileUserPasswdStore.resolveFile(env);
+        final Path rolesFile = FileUserRolesStore.resolveFile(env);
+        FileAttributesChecker attributesChecker = new FileAttributesChecker(passwordFile, rolesFile);
+
+        Map<String, char[]> users = FileUserPasswdStore.parseFile(passwordFile, null, env.settings());
+        if (users == null) {
+            throw new UserException(ExitCodes.CONFIG, "File realm configuration file [" + passwordFile + "] is missing");
+        }
+        users = new HashMap<>(users);
+        char[] passwd = users.remove(username);
+        if (passwd != null) {
+            // No need to overwrite, if the user was already removed
+            FileUserPasswdStore.writeFile(users, passwordFile);
+            Arrays.fill(passwd, '\0');
+        }
+
+        Map<String, String[]> userRoles = FileUserRolesStore.parseFile(rolesFile, null);
+        if (userRoles == null) {
+            throw new UserException(ExitCodes.CONFIG, "File realm configuration file [" + rolesFile + "] is missing");
+        }
+        userRoles = new HashMap<>(userRoles);
+        String[] roles = userRoles.remove(username);
+        if (roles != null) {
+            // No need to overwrite, if the user was already removed
+            FileUserRolesStore.writeFile(userRoles, rolesFile);
+        }
+        attributesChecker.check(terminal);
+    }
+
+    protected SecureString getPassword() {
+        return password;
+    }
+
+    protected String getUsername() {
+        return username;
+    }
+
+    private void ensureFileRealmEnabled(Settings settings) throws Exception {
+        Map<String, Settings> fileRealmSettings = settings.getGroups("xpack.security.authc.realms.file");
+        if (fileRealmSettings.size() > 1) {
+            throw new UserException(
+                ExitCodes.CONFIG,
+                "Multiple file realms are configured. "
+                    + "[file] is an internal realm type and therefore there can only be one such realm configured"
+            );
+        } else if (fileRealmSettings.size() == 1
+            && fileRealmSettings.entrySet().stream().anyMatch(s -> s.getValue().get("enabled").equals("false"))) {
```

Comment:
If it's possible to use `RealmConfig` for this (rather than having a variety of code that knows about the `enabled` setting) that would be helpful.

Reply:
I adjusted this a little to be less custom. I looked into using `RealmConfig.enabled()` but I'd need to iterate through all realms first to get the name of the file realm in order to build the RealmIdentifier that I need to get the RealmConfig object , so that I can call `enabled()` on it. Let me know if the current version is more passable

Reply:
What you have is fine, but I was thinking more like what `InternalRealms.getBootstrapChecks` does.

Reply:
Now that I saw that, it clicked